packages/kbn-elastic-agent-utils/src/agent_guards.ts

```diff
@@ -7,31 +7,52 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { JAVA_AGENT_NAMES, OPEN_TELEMETRY_AGENT_NAMES, RUM_AGENT_NAMES } from './agent_names';
+import {
+  ANDROID_AGENT_NAMES,
+  IOS_AGENT_NAMES,
+  JAVA_AGENT_NAMES,
+  OPEN_TELEMETRY_AGENT_NAMES,
+  RUM_AGENT_NAMES,
+} from './agent_names';
 
 import type {
+  AndroidAgentName,
+  IOSAgentName,
   JavaAgentName,
   OpenTelemetryAgentName,
   RumAgentName,
   ServerlessType,
 } from './agent_names';
 
+export function hasOpenTelemetryPrefix(agentName?: string, language: string = '') {
+  if (!agentName) {
+    return false;
+  }
+
+  return (
+    agentName.startsWith(`opentelemetry/${language}`) || agentName.startsWith(`otlp/${language}`)
```

Comment:
Nit:
```suggestion
    agentName.startsWith(`opentelemetry/${language}` || `otlp/${language}`);
```

Reply:
I think the proposed solution does not apply here, as `||` acts as a fallback value if the first one is falsy, but as it isn't it's not going to work, the function is not able to know which value should compare it to, so it does it for the first one always.

See the example below
![image](https://github.com/user-attachments/assets/9aa19525-af42-4cc6-a814-a4bd62ce1b62)